app/src/main/java/com/example/homework1calculator/Calculator.kt

```diff
@@ -0,0 +1,174 @@
+package com.example.homework1calculator
+
+
+class Calculator {
+
+    private val regFindNum = Regex("[-*/+]")
+    private val regFindSigns = Regex("[^-+*/=]")  //(?<=[^-+*/=,{ ])[-+*/=]
+    private val regexSpace = Regex("[\\s]")
+    private val priorityMap = mapOf(
+
+        "+" to 1,
+        "-" to 1,
+        "*" to 2,
+        "/" to 2
+    )
+
+    fun calculate(text: String?): String {
+
+        return text?.let {          // функция расширения .let http://developer.alexanderklimov.ru/android/kotlin/with-apply-also.php
+            if (it.contains('(')) { //
+                calcWithBraces(it)
+            } else {
+                calculateWithNoBraces(it)
+            }
+        } ?: ""
+    }
```

Comment:
в целом не сильно вижу смысла задавать такое поведение калькулятора чтобы туда можно было null передавать
конкретно в твоем случае можно было еще сделать так, будет меньше вложенность скобок, и тут также потом компилятор приведет переменную к ненулевой
```suggestion
    fun calculate(text: String?): String {
        text ?: return ""
        return if (text.contains('(')) { 
            calcWithBraces(it)
        } else {
            calculateWithNoBraces(it)
        }
    }
```